utils.js

```diff
@@ -801,14 +894,110 @@ HDSigner.prototype.signPSBT = async function (psbt) {
+  trezortx.inputs = [];
+  trezortx.outputs = [];
+  
+  for (let i = 0; i < psbt.txInputs.length; i++) {
+      let script_types = psbt.getInputType(i);
+      const input = psbt.txInputs[i];
+      const inputItem = {};
+      inputItem.prev_index = input.index;
+      inputItem.prev_hash = input.hash.reverse().toString('hex')
+      if (input.sequence) inputItem.sequence = input.sequence;
+      const dataInput = psbt.data.inputs[i]
+      let path = ''
+      if (dataInput.unknownKeyVals && dataInput.unknownKeyVals.length > 1 && 
+              dataInput.unknownKeyVals[1].key.equals(Buffer.from('path')) && 
+                      (!dataInput.bip32Derivation || dataInput.bip32Derivation.length === 0)) {
+        path = (dataInput.unknownKeyVals[1].value.toString())
+        inputItem.address_n = convertToAddress_nFormat(path)
+      }
+      script_types = (Number(path[2] + path[3]));
+      switch (script_types) {
+        case 48:
+          inputItem.script_type =  'SPENDMULTISIG';
+          break;
+        case 49:
+          inputItem.script_type =  'SPENDP2SHWITNESS';
+          break;
+        case 84:
+          inputItem.script_type =  'SPENDWITNESS';
+          break;
+        default:
+          inputItem.script_type =  'SPENDADDRESS';
+          break;
+    }
+    trezortx.inputs.push(inputItem);
+  }
+  
+  for(let i = 0; i < psbt.txOutputs.length; i++){
+      const output = psbt.txOutputs[i];
+      const outputItem = {};
+      const chunks = bjs.script.decompile(output.script);
+      outputItem.amount = output.value.toString();
+      if (chunks[0] === bitcoinops.OP_RETURN) {
+        outputItem.script_type = 'PAYTOOPRETURN';
+        outputItem.op_return_data = chunks[1].toString('hex');
+      }
+      else{
+        if(isBech32(output.address)){
+          outputItem.script_type = 'PAYTOWITNESS'
+        }
+        else{
+          outputItem.script_type = isScriptHash(output.address, this.network) ? 'PAYTOSCRIPTHASH' : 'PAYTOADDRESS';
+        }
+        outputItem.address = output.address;
+      }
+      trezortx.outputs.push(outputItem);
+    }
+    console.log(trezortx)
+    return trezortx;
+  }
+
+
+
+/* sign
+Purpose: Create signing information based on Trezor format
+Param psbt: Required. PSBT object from bitcoinjs-lib
+Param getTezortx: Optional. If true, will return trezor params to signTransaction and wont signAndSend it
+Returns: trezortx or txid
+*/
+TrezorSigner.prototype.sign = async function (psbt, getTezortx) {
+  
+  if(psbt.txInputs.length <=0 || psbt.txOutputs.length <=0 || psbt.version === undefined){
+    throw new Error('PSBT object is lacking information')
+  }
+  const sendTrezorTx = getTezortx || false
+  const trezorTx = this.convertToTrezorFormat(psbt)
+  if(!sendTrezorTx){
+    const response = await TrezorConnect.signTransaction(trezorTx)  
+    if(response.success === true){
+      txInfo = await sendRawTransaction(this.blockbookURL, response.payload.serializedTx)
```

Comment:
does sign usually send? it should follow the same logic as hdsigner sign, sending might be seperate